internal/data/test/load.go

```diff
@@ -9,18 +9,19 @@ import (
 	"strings"
 
 	"github.com/pkg/errors"
+	"github.com/wallarm/gotestwaf/internal/data/config"
 	"gopkg.in/yaml.v2"
 )
 
-func Load(testCasesPath string, logger *log.Logger) ([]Case, error) {
+func Load(cfg *config.Config, logger *log.Logger) ([]Case, error) {
 	var files []string
 	var testCases []Case
 
-	if testCasesPath == "" {
+	if cfg.TestCasesPath == "" {
 		return nil, errors.New("empty test cases path")
 	}
 
-	if err := filepath.Walk(testCasesPath, func(path string, info os.FileInfo, err error) error {
+	if err := filepath.Walk(cfg.TestCasesPath, func(path string, info os.FileInfo, err error) error {
```

Comment:
Test cases (files) handling doesn't work properly if we have custom test cases path:
```
➜  gotestwaf git:(httpworkers) ✗ go run main.go --testCasesPath ./subdir/subsubdir/testcases/ --url http://localhost:8080/   
GOTESTWAF : 2021/01/29 00:56:36.063416 cmd.go:47: Test cases loading started
GOTESTWAF : 2021/01/29 00:56:36.063840 load.go:31: Loading test cases: 
GOTESTWAF : 2021/01/29 00:56:36.063854 load.go:41: subsubdir:testcases
GOTESTWAF : 2021/01/29 00:56:36.063949 load.go:41: subsubdir:testcases
GOTESTWAF : 2021/01/29 00:56:36.064039 load.go:41: subsubdir:testcases
GOTESTWAF : 2021/01/29 00:56:36.064161 load.go:41: subsubdir:testcases
GOTESTWAF : 2021/01/29 00:56:36.064605 load.go:41: subsubdir:testcases
GOTESTWAF : 2021/01/29 00:56:36.064664 load.go:41: subsubdir:testcases
GOTESTWAF : 2021/01/29 00:56:36.064726 load.go:41: subsubdir:testcases
GOTESTWAF : 2021/01/29 00:56:36.064778 load.go:41: subsubdir:testcases
GOTESTWAF : 2021/01/29 00:56:36.064858 load.go:41: subsubdir:testcases
GOTESTWAF : 2021/01/29 00:56:36.064978 load.go:41: subsubdir:testcases
GOTESTWAF : 2021/01/29 00:56:36.065049 load.go:41: subsubdir:testcases
GOTESTWAF : 2021/01/29 00:56:36.065104 load.go:41: subsubdir:testcases
GOTESTWAF : 2021/01/29 00:56:36.065160 load.go:41: subsubdir:testcases
GOTESTWAF : 2021/01/29 00:56:36.065213 load.go:41: subsubdir:testcases
GOTESTWAF : 2021/01/29 00:56:36.065281 load.go:41: subsubdir:testcases
GOTESTWAF : 2021/01/29 00:56:36.065370 load.go:41: subsubdir:testcases
GOTESTWAF : 2021/01/29 00:56:36.065428 load.go:41: subsubdir:testcases
GOTESTWAF : 2021/01/29 00:56:36.065480 load.go:41: subsubdir:testcases
GOTESTWAF : 2021/01/29 00:56:36.065532 load.go:41: subsubdir:testcases
GOTESTWAF : 2021/01/29 00:56:36.065580 cmd.go:53: Test cases loading finished
GOTESTWAF : 2021/01/29 00:56:36.065597 cmd.go:59: Scanned URL: http://localhost:8080/
GOTESTWAF : 2021/01/29 00:56:36.074085 cmd.go:72: WAF pre-check: OK. Blocking status code: 403
GOTESTWAF : 2021/01/29 00:56:36.074388 cmd.go:91: Scanning http://localhost:8080/
GOTESTWAF : 2021/01/29 00:56:36.074417 scanner.go:80: Scanning started
GOTESTWAF : 2021/01/29 00:56:39.282486 scanner.go:85: Scanning Time:  3.208000017s
GOTESTWAF : 2021/01/29 00:56:39.282508 scanner.go:116: Scanning finished
+-----------+-----------+--------+----------------+-----------------+
| TEST SET  | TEST CASE | PASSED | PASSED/BLOCKED | FAILED/BYPASSED |
+-----------+-----------+--------+----------------+-----------------+
| subsubdir | testcases |  63.11 |            378 |             221 |
+-----------+-----------+--------+----------------+-----------------+
|                                    WAF SCORE:   |     63.11%      |
+-----------+-----------+--------+----------------+-----------------+
```
Any user-defined path here will break the test cases loading.

The problem is here - indexes are fixed for the path:
```go
		parts := strings.Split(testCaseFile, "/")
		testSetName := parts[1]
		testCaseName := strings.TrimSuffix(parts[2], path.Ext(parts[2]))
```
We need to look at them from the end of the path, not with fixed numbers from the beginning because the path can be long/short/anything.